### ISSUE TITLE
run_7point returns solve_cubic's padded zero roots as duplicate rank-3 candidates; its validity mask never zeroes anything

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4862]
`run_7point` (and so `find_fundamental(..., method="7POINT")`) returns every root that `solve_cubic` produces as a fundamental-matrix candidate. `solve_cubic` reports only real roots and pads the missing ones with `0.0`. When the 7-point cubic has one real root — 49 of 300 random exact samples, about 16% — the two padded zeros become two identical candidates `F = f2 / f2[2, 2]`, the second null-space vector, which is rank 3 and satisfies the epipolar constraint only on the seven sample points. They are returned un-flagged: the docstring says "Invalid solutions are zeroed out", but the mask `valid_root_mask = (cnz < 3) | (cnz > 1)` is true for every value of `cnz`, so nothing is ever zeroed.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import projection_from_KRt
from kornia.geometry.epipolar.fundamental import run_7point

torch.manual_seed(1)
h = lambda p: torch.cat([p, torch.ones_like(p[..., :1])], -1)
def proj(P, X):
    x = (P @ h(X).transpose(1, 2)).transpose(1, 2)
    return x[..., :2] / x[..., 2:]

K1 = torch.tensor([[[800.0, 0, 320], [0, 760, 200], [0, 0, 1]]])
K2 = torch.tensor([[[700.0, 0, 300], [0, 740, 240], [0, 0, 1]]])
dup = 0
for i in range(300):
    R = axis_angle_to_rotation_matrix(torch.randn(1, 3) * 0.2)
    t = torch.randn(1, 3, 1)
    X = torch.rand(1, 7, 3) * 2 - 1
    X[..., 2] = 4 + 2 * torch.rand(1, 7)
    P1 = projection_from_KRt(K1, torch.eye(3)[None], torch.zeros(1, 3, 1))
    P2 = projection_from_KRt(K2, R, t)
    Fs = run_7point(proj(P1, X), proj(P2, X))[0]          # (3, 3, 3)
    if torch.allclose(Fs[1], Fs[2], atol=1e-6) and not torch.allclose(Fs[0], Fs[1], atol=1e-6):
        dup += 1
        if dup == 1:
            print("cubic had one real root; candidates 2 and 3 are the same matrix, det =",
                  torch.linalg.det(Fs[1]).item())   # ~1e-3 on the F[2,2]=1 scale: rank 3, not a fundamental matrix
print(dup, "of 300 samples")                            # 49 of 300
```

kornia `main` 7ddf731bd, torch 2.14.0, CPU float32, macOS arm64. Spying on `solve_cubic` inside `run_7point` confirms every duplicate case has roots `[r, 0.0, 0.0]` with a negative cubic discriminant and a constant term `det(f2)` of order 1e-3, so the zeros are padding, not roots.

## Classification

Clearly broken output. `find_fundamental(method="7POINT")` promises up to three fundamental matrices with the invalid ones zeroed; for a one-real-root sample it returns one valid matrix and two copies of a rank-3 matrix that is not one. `RANSAC(model_type="fundamental_7pt")` scores and discards them, so its result is unaffected and one third of its candidate budget is wasted; a direct caller has no way to tell the padded candidates from real ones.

Idiom 4 (degenerate output indistinguishable from valid output). The root cause is the design choice, ruled on 2026-09-24 in the conventions batch 8 audit, that `solve_quadratic` / `solve_cubic` / `solve_quartic` report real roots only and pad the rest with `0.0`; the consumer has to count real roots itself and does not.

## Disposition

Focused fix, welcome as a PR: compute the number of real roots from the cubic's discriminant (or from `solve_cubic`'s own branch), and zero the candidates for the padded roots so the documented "Invalid solutions are zeroed out" contract holds; a `test_wart_*` pin in the batch 8 docs PR flips when it lands. Not a window item: no user-visible convention changes, the docstring's contract becomes true.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
